src/goto-programs/json_goto_trace.cpp

```diff
@@ -141,14 +145,22 @@ void convert(
         json_output["thread"]=json_numbert(std::to_string(step.thread_nr));
         json_output["outputID"]=json_stringt(id2string(step.io_id));
 
+        // Recovering the mode from the function
+        irep_idt mode;
+        const symbolt *function_name;
+        if(ns.lookup(source_location.get_function(), function_name))
+          // Failed to find symbol
+          mode=ID_unknown;
+        else
+          mode=function_name->mode;
         json_arrayt &json_values=json_output["values"].make_array();
 
         for(const auto &arg : step.io_args)
         {
           if(arg.is_nil())
             json_values.push_back(json_stringt(""));
```

Comment:
My guess is that something needs to be pushed here so that the numbering of array elements is the same as that of the arguments.